test/minikube/minikube_base_insights_test.go

```diff
@@ -0,0 +1,200 @@
+// +build long
+
+package minikube
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/nuodb/nuodb-helm-charts/test/testlib"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	v12 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const FILE_PLUGIN_CONFIGMAP = `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nuocollector-insights-file
+  labels:
+    nuodb.com/nuocollector-plugin: insights
```

Comment:
I will comment here, but it might apply to other places. I am not sure whether "insights" here is the right thing. I hope that it is not hardcoded. I assume that we pick up any configmap that has the `nuodb.com/nuocollector-plugin` label. What do we do with the value of the label?

Reply:
I believe it is hardcoded and we do nothing with the value just the key.  If it was configurable we could change the k8s-sidecar configuration to select based upon LABEL_VALUE allowing us to deploy different dynamic plugins to different components (admin or database) or different helm deployments.

Reply:
>I assume that we pick up any configmap that has the nuodb.com/nuocollector-plugin label.
Nope. We pick up any configMap that match all of the following criteria:
* in the same namespace
* has a label of `nuodb.com/nuocollector-plugin`
* The value of the label is either `insights` OR specified in `nuocollector.plugins.admin` or `nuocollector.plugins.database`.

To be more specific the configMap selector is:
* for admin plugins: `"nuodb.com/nuocollector-plugin in ({{ template "admin.fullname" $ }}, insights)"`
* for database plugins: `"nuodb.com/nuocollector-plugin in ({{ template "database.fullname" $ }}, insights)"`

>If it was configurable we could change the k8s-sidecar configuration to select based upon LABEL_VALUE allowing us to deploy different dynamic plugins to different components (admin or database) or different helm deployments.

You can still deploy dynamic plugins for different components/deployments --- see https://github.com/nuodb/nuodb-collector#deploying-custom-plugins. I don't really see a reason for making the "insights" label configurable because there is no need for the user to change this. I personally would like users to use the helm charts to deploy new plugins vs creating configMaps by hand.